Serilog.Sinks.Network/Sinks/TCP/TCPSocketWriter.cs

```diff
@@ -136,7 +138,7 @@ public TcpSocketWriter(Uri uri, int maxQueueSize = 5000)
                             try
                             {
                                 var messsage = Encoding.UTF8.GetBytes(entry);
-                                if (uri.Scheme.ToLower() == "tls")
+                                if (sslEnabled)
                                     ((SslStream)_stream).Write(messsage);
                                 else
                                     _stream.Write(messsage, 0, messsage.Length);

```

Comment:
Not sure why the SSL case was separated, the common Stream.WriteAsync() method looks to be suitable. I'll run some SSL tests with the new code once I get to the office.